src/chunk_cache.rs

```diff
@@ -4,11 +4,28 @@ use crate::error::ActiveStorageError;
 use byte_unit::Byte;
 use bytes::Bytes;
 use serde::{Deserialize, Serialize};
-use std::{collections::HashMap, ops::Add, path::PathBuf, time::{SystemTime, UNIX_EPOCH}};
-use tokio::fs;
+use std::{collections::HashMap, ops::Add, path::PathBuf, sync::Arc, time::{SystemTime, UNIX_EPOCH}};
+use tokio::{fs, sync::mpsc, spawn};
+
+struct KeyValueMessage {
+    key: String,
+    value: Bytes,
+}
+
+impl KeyValueMessage {
+    fn new(key: String, value: Bytes) -> Self {
+        // Make sure the message owns the Bytes so we don't see unexpected, but not incorrect, behaviour caused by zero copy.
+        let value = Bytes::from(value.to_vec());
+        Self {
+            key,
+            value,
+        }
+    }
+}
 
 pub struct ChunkCache {
-    cache: SimpleDiskCache,
+    cache: Arc<SimpleDiskCache>,
+    sender: mpsc::Sender<KeyValueMessage>,
```

Comment:
Would be good to add some doc comments here, e.g.
```suggestion
    /// The underlying data cache
    cache: Arc<SimpleDiskCache>,
    /// Sync primitive for managing write access to the cache
    sender: mpsc::Sender<KeyValueMessage>,
```